### PR TITLE
Condec 503.rest.api.feature.branch.rationale

### PR DESCRIPTION
**Pull request by lw2011**
Hi @kleebaum and @dekome ,

here is a tiny branch introducing new rest api method.

Please note the rationale documented in commit's 0fe070c message. I have deliberately picked the wrong alternative as decision, which should not be picked as one.

EDIT: tests added for new classes

### CODE REVIEW

File: src/main/java/de/uhd/ifi/se/decision/management/jira/extraction/impl/GitClientImpl.java [hunk 1]
```diff
@@ -468,9 +468,21 @@ public boolean checkoutFeatureBranch(Ref featureBranch) {
...
 		}
-		// @issue How to get the commits for branches that are not on the master branch?
-		// @decision Assume that the JIRA issue key equals the branch name, otherwise
-		// return commits on master branch
+		/**
+		 * @issue How to get the commits for branches that are not on the master branch?
+		 * @decision Assume that the branch name begins with the JIRA issue key!
+		 * @pro This simple rule will work just fine.
+		 * @con The rule is too simple. Issues with low key numbers could collect branches
+		 * of much higher issues also. ex. search for "CONDEC-1" would find branches
```
Comment (kleebaum) [inline src/main/java/de/uhd/ifi/se/decision/management/jira/extraction/impl/GitClientImpl.java]: We could overcome the problem that looking for CONDEC-1 could also return CONDEC-1x with the rule that there needs to be a colon, dot or space after the issue key. (The space is not allowed in branch names, but in commit messages.)

Comment (kleebaum) [reply src/main/java/de/uhd/ifi/se/decision/management/jira/extraction/impl/GitClientImpl.java]: You stated this below, sorry ;-)

File: src/main/java/de/uhd/ifi/se/decision/management/jira/view/diffviewer/BranchDiff.java [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+
+import de.uhd.ifi.se.decision.management.jira.model.DecisionKnowledgeElement;
+
+import javax.xml.bind.annotation.XmlElement;
+import java.util.*;
+
+/**
+ * Model class for DiffViewer's branches node
+ */
+public class BranchDiff {
```
Comment (kleebaum) [inline src/main/java/de/uhd/ifi/se/decision/management/jira/view/diffviewer/BranchDiff.java]: This class might be better placed in a different package than the view package, e.g. the model package. We should think of adding more model classes for diffs and changed files.

Comment (lw2011) [reply src/main/java/de/uhd/ifi/se/decision/management/jira/view/diffviewer/BranchDiff.java]: Yes, I was also thinking about that, but for the beginning I chose to follow the (no so intuitive) conventions used in the project for treant, treview, vis..

I will refactor this in a new branch, as this problem resolution could be applied to above mentioned classes too.

**Comment by lw2011** [comment]
CI tool fails to build with error "TestElementsFromBranchesOfJiraIssue.testExistingIssueKey:61 » NullPointer" which does not happen for local builds on my machine. Does Travis have issues with overwriting branches?, but the logs suggest otherwise.

**Pull request merged**